### DISCUSSION TITLE
Prompt Lists, Filenames in outputs (Of prompt used)

### DISCUSSION THREAD

**Discussion by cryptodeveloperbro** [comfyanonymous/ComfyUI#985] [Ideas]
2 Major things missing from Comfy.  I am wondering if the ability to [prompt_spaces] like on Auto1111 will be added to it.  Not saving the image generation with the prompt as a filename is strange to me.  Seeing ComfyUI_000108 for filenames is not ideal.  The prompt should be in the filename like Auto1111.  And then, adding a list of Prompts in a box will be helpful as well.  That way we can put a list of 50-100 prompts or however many we want.  Then we can set it, set the batch amount and go to sleep while it generates.  Anyway, if these features are coming soon that will be great!   I hear using SDXL in Auto1111 is way slower so I have not tried it yet.   Are theses incoming features?  TIA.

**Comment by pvp-by** [comment]
comprehension difficulty medium :
https://blenderneko.github.io/ComfyUI-docs/Interface/SaveFileFormatting/

**Comment by dievirulenz** [comment]
[EDIT]
Okay I dove into it a bit more, and got a better understanding. I now use the "Clip Text Encode (Prompt)" and its Text widget to put into the output filename.

---

I was also searching for this.

What I dont understand is how to get a useable node name from a primitive node to feed into the output string?

Like I got the primitive node with the input text and want to reference that for the output string. But I cant seem to edit the nodes "name for S&R". Also not sure if the textfield is considered a widget?

I'd be grateful for any tips. Many thanks.

**Comment by azureprophet** [comment]
Also, is there a function to automatically make a folder with the first x terms in a prompt for each batch?
